### DISCUSSION TITLE
Investigating invalid uri character

### DISCUSSION THREAD

**Discussion by troelsarvin** [sagebind/isahc#347] [Questions]
I'm fetching a lot of rather long URLs like this:
http://somewhere/xxxx/yyy/zzz?variable1=1.2.6.1.5.1.99999.2.11614477878827115943.6363256.27342.4221&someparameter=SOMEVALUE&somevariable=someratherlongvalue

For a large subset of my requests, my get()-call result in an error which I grab like this:
  warn!("Error GETing URL {}: {}", url, e.source().unwrap_or(&e));
The warning I get is: invalid uri character

But there doesn't seem to be anything wrong with the URLs. At some point, I had my code run 
  let _urlcheck = Url::parse(url.as_str()).unwrap();
on URLs before using them with isahc::get(), and Url::parse didn't reject any of the URLs.

How can I dig deeper in to why isahc thinks the URL contains an invalid charter?

**Comment by sagebind** [comment]
Internally URL parsing is done with [`Uri`](https://docs.rs/http/0.2.4/http/uri/struct.Uri.html) from the `http` crate, so maybe there's something in your URLs that it doesn't like? You could try parsing your URLs with that and see if you can get any more details. If you could share a specific URL that causes that error that would be helpful.
